frontend/src/pages/PipelineDetails.test.tsx

```diff
@@ -536,34 +529,51 @@ describe('PipelineDetails', () => {
     );
   });
 
-  it('shows load error banner when failing to get pipeline template', async () => {
-    TestUtils.makeErrorResponseOnce(getPipelineVersionTemplateSpy, 'woops');
-    tree = shallow(<PipelineDetails {...generateProps()} />);
-    await getV1PipelineSpy;
-    await TestUtils.flushPromises();
-    expect(updateBannerSpy).toHaveBeenCalledTimes(2); // Once to clear banner, once to show error
+  it('shows load error banner when failing to get pipeline version', async () => {
+    TestUtils.makeErrorResponse(getV2PipelineVersionSpy, 'No pipeline version is found');
+    render(<PipelineDetails {...generateProps()} />);
+
+    await waitFor(() => {
+      // one for selected Version, another for template string
+      expect(getV2PipelineVersionSpy).toHaveBeenCalledTimes(2);
```

Comment:
Can we improve this by only querying the same API once? It feels strange to me that `updateBanner` will be called twice in the following validation. It is probably a bug that needs fixing because banner has been updated twice.